### ISSUE TITLE
15_autoencoders - Unsupervised pretraining is not implemented as book figure 15-8

### ISSUE THREAD

**Issue by ericchencth** [ageron/handson-ml#35]
hi , @ageron 
in the book figure 15-8:  "_you can first train a stacked autoencoder using all the data, then reuse the lower layers to create a neural network for your actual task, and train it using the labeled data_"

but in the example code phase 1 , it  does not use all the data  "output ≈ input" to train the autoencoder.

Thanks!

**Comment by ageron** [comment]
Hi @ericchencth , thanks for your feedback.

Suppose you have 1 million images of handwritten digits, but only 70,000 are labeled. You could train the autoencoder on the full one million images, then reuse the lower layers of the autoencoder to build a new neural net for classification, train it on 60,000 labeled instances, and test it on the remaining 10000.  Unfortunately, I don't actually have a million unlabeled digit images to demonstrate this, so I trained the autoencoder on the same 60,000 labeled instances.  Perhaps that's what's confusing.  Maybe I should split the 60,000 training instances in two: say, 40,000 instances for which I would drop the label, just for the purpose of the demonstration, and 20,000 for training the classifier.

What do you think?

**Comment by ericchencth** [comment]
hi @ageron ,
Thanks for your quick responding!
I  misunderstand the code below:
      pretrain_saver.restore(sess, "./my_model_cache_frozen.ckpt")
This saver is from the 'Cache the frozen layer outputs' paragraph, not from the code above this cell.
So this code is absolutely correct 👍 
Thanks again!
